lib/src/metta/runner/mod.rs

```diff
@@ -241,11 +241,43 @@ impl Metta {
         self.0.settings.borrow().get(key.into()).map(|a| a.to_string())
     }
 
+    /// Runs a MeTTa program expressed as a string of characters
+    pub fn run_program_str(&self, program: &str) -> Result<Vec<Vec<Atom>>, String> {
+        let parser = SExprParser::new(program);
+        self.run(parser)
+    }
+
     pub fn run<'p, 'a: 'p>(&'a self, parser: SExprParser<'p>) -> Result<Vec<Vec<Atom>>, String> {
         let state = RunnerState::new_with_parser(self, parser);
         state.run_to_completion()
     }
 
+    /// Parses a string into a single Atom
+    pub fn parse_one_atom(&self, text: &str) -> Result<Atom, String> {
+        let mut parser = SExprParser::new(text);
+        let atom = parser.parse(&*self.tokenizer().borrow())?;
+        if let Some(atom) = atom {
+            Ok(atom)
+        } else {
+            panic!("Expected a single atom");
+        }
+    }
+
+    /// Parses a string into a vector of Atoms
+    pub fn parse_all_atoms(&self, text: &str) -> Result<Vec<Atom>, String> {
```

Comment:
This method has the only usage and can be implemented there.

Reply:
This is a nuanced point, but here is my thinking:

1. An interpreter acts on just atoms.  Parsing happens before the interpreter.
2. It makes sense to have the ability to run raw atoms, rather than forcing the user of the runner to first render them to text and re-parse them.  This is done with `RunnerState::new_with_atoms()`
3. This function (`Metta::parse_all_atoms()`) is the other half of that flow.  It lets the caller parse a string of MeTTa code as the runner would.

So, why isn't `parse_all_atoms` a `SExprParser` method, instead of the `Metta` method?
1. The parser has an implicit dependency on a certain Tokenizer, in order to create the right atoms from the input text.
2. Tokens get loaded into the Tokenizer as an effect of running MeTTa code (e.g. `bind!`) or loading modules, and therefore the Tokenizer is part of the Runner's state.

Reply:
Ok, after the discussion last night, I removed Metta::parse_all_atoms and Metta::parse_one_atom in 28e464e

Aesthetically I feel like this is not great because parsing atoms requires a tokenizer, which is initialized through a runner, and borrowing a runner's tokenizer is very ugly.

So a conceptually straightforward operation, (parsing MeTTa code into an Atom) the following ugly Rust code:
```Rust
SExprParser::new("(Cons 1 Nil)").parse(&*metta.tokenizer().borrow())?.unwrap()
```
This would be a lot nicer if `Metta::tokenizer` could return an ordinary reference.

Reply:
Thanks, Luke